routing/dht/routing.go

```diff
@@ -153,7 +168,7 @@ func (dht *IpfsDHT) Provide(ctx context.Context, key u.Key) error {
 // FindProviders searches until the context expires.
 func (dht *IpfsDHT) FindProviders(ctx context.Context, key u.Key) ([]peer.PeerInfo, error) {
 	var providers []peer.PeerInfo
-	for p := range dht.FindProvidersAsync(ctx, key, math.MaxInt32) {
+	for p := range dht.FindProvidersAsync(ctx, key, KValue) {
```

Comment:
why `KValue` and not max until it expires?  (just for allocation? is there another workaround?) will this break something else?